src/laptools/clap.py

```diff
@@ -146,16 +167,33 @@ def costs(cost_matrix):
                 col_idxs[other_i] = next_best_j
                 total_costs[i, stolen_j] = cost_matrix[row_idxs, col_idxs].sum()
             else:
-                sub_cost_matrix = cost_matrix[:, potential_cols]
-                sub_j = np.argwhere(potential_cols == stolen_j)[0]
-                total_cost = cost(i, sub_j, sub_cost_matrix)
-                total_costs[i, stolen_j] = total_cost
+                sub_lsap_row4col = new_lsap_row4col.copy()
+                # Note that row i is assigned to some column. Need to remove
+                # that from the assignment.
+                new_freed_j = new_lsap_col4row[i]
+                sub_lsap_row4col[new_freed_j] = -1
+                sub_lsap_row4col[sub_lsap_row4col > i] -= 1
```

Comment:
Is any of this things that could be done outside the loop?

Reply:
Noted. Will fix.